ci/wait-for-statuses.py

```diff
@@ -49,13 +36,11 @@
 scripts_dir = os.environ['CI_SCRIPTS_PATH']
 
 branch = os.environ.get('GITHUB_REF', '')
-# Upload packages only when whole build succeeded
-# and we are on master branch
-if(branch == 'refs/heads/master'):
-  if(not jobFailure):
-    subprocess.call(os.path.join(scripts_dir, "master-package.sh"))
-else:
+# Upload packages only when whole build succeeded and we are on master branch
+if not (branch == 'refs/heads/master'):
   print("Not on master branch, don't execute master-package.sh. Current branch: " + str(branch))
+elif(not jobFailure):
+  subprocess.call(os.path.join(scripts_dir, "master-package.sh"))
 
 # Always clean up
 subprocess.call(os.path.join(scripts_dir, "cleanup-anaconda.sh"))
```

Comment:
This is a minor drawback of the change, the cleaning script will now only be run if all the packages were successfully built. Perhaps you could extract it to a separate job if it's not too much work otherwise just create an issue to remember about it and remove the comment above.

Reply:
I'm not sure I understand the proposed change. I did two modifications:

- Made the Upload job run unconditionally (`always()`).
- Cleaned the `wait-for-statuses` to execute `master-package.sh` regardless of job failures.

Let me know whether that's what you expected.